mongo-stream.js

```diff
@@ -0,0 +1,277 @@
+const BSON = require('bson');
+const bson = new BSON();
+const elasticsearch = require('elasticsearch');
+const fs = require('fs');
+const MongoClient = require('mongodb').MongoClient;
+
+
+class MongoStream {
+  constructor(esClient, db) {
+    this.esClient = esClient;
+    this.db = db;
+    this.changeStreams = {};
+  }
+
+  // constructs and returns a new MongoStream
+  static async init() {
+    const { url, mongoOpts } = MongoStream.setMongoOpts();
+    const dbName = process.env.MONGO_DB;
```

Comment:
You should avoid commingling external configuration details and the class being configured. By forcing environmentals on the primary entry point, you make out of band communication part of the API.